core/arch/arm/kernel/generic_boot.c

```diff
@@ -1106,6 +1130,13 @@ static void init_primary_helper(unsigned long pageable_part,
 	init_tee_runtime();
 #endif
 	release_external_dt();
+#ifdef CFG_OVERLAY_ADDR
+	dt->is_overlay = 1;
+	init_external_dt(CFG_OVERLAY_ADDR);
+	discover_nsec_memory();
+	update_external_dt();
+	release_external_dt();
+#endif
```

Comment:
I'd rather have a helper function for this to improve legibility and avoid the `#ifdef`